pages/team-members/profile/[mail].tsx

```diff
@@ -0,0 +1,61 @@
+import { Divider, Grid, Paper } from '@mui/material';
+import CustomError from '../../../components/CustomError';
+import { getSession } from 'next-auth/react';
+import prisma from '../../../lib/prisma';
+import Profile from '../../../components/team-members/Profile';
+
+export async function getServerSideProps({ params, req }) {
+  const session = await getSession({req})
+  if (!session) {
```

Comment:
is there a way that we can have a wrapper component that checks the permissions in order to avoid the repetitive code?